src/osemgrep/cli_login/Test_login_subcommand.ml

```diff
@@ -119,45 +100,49 @@ let fake_deployment =
+            "You're already logged in";
+            "Logged out!";
+            "You are not logged in";
+          ];
+      ]
+    __FUNCTION__
     (with_login_test_env (fun () ->
          Semgrep_envvars.with_envvar "SEMGREP_APP_TOKEN" fake_token (fun () ->
              with_fake_deployment_response fake_deployment (fun () ->
-                 (* login with env token *)
-                 with_logs
-                   ~f:(fun () ->
-                     Login_subcommand.main caps [| "semgrep-login" |])
-                   ~final:(fun res ->
-                     assert (res.logs =~ "[.\n]*Saved access token");
-                     assert (res.exit_code =*= Exit_code.ok));
-
-                 (* login should fail on second call *)
-                 with_logs
-                   ~f:(fun () ->
-                     Login_subcommand.main caps [| "semgrep-login" |])
-                   ~final:(fun res ->
-                     assert (res.logs =~ ".*You're already logged in");
-                     assert (res.exit_code =*= Exit_code.fatal));
-
-                 (* clear login (by logging out) *)
-                 with_logs
-                   ~f:(fun () ->
-                     Logout_subcommand.main
-                       (caps :> < Cap.stdout >)
-                       [| "semgrep-logout" |])
-                   ~final:(fun res ->
-                     assert (res.logs =~ ".*Logged out!");
-                     assert (res.exit_code =*= Exit_code.ok));
-
-                 (* logout twice should work *)
-                 with_logs
-                   ~f:(fun () ->
-                     Logout_subcommand.main
-                       (caps :> < Cap.stdout >)
-                       [| "semgrep-logout" |])
-                   ~final:(fun res ->
-                     assert (res.logs =~ ".*You are not logged in");
-                     assert (res.exit_code =*= Exit_code.ok))))))
+                 let exit_code =
+                   Login_subcommand.main caps [| "semgrep-login" |]
+                 in
+                 (* assert (res.logs =~ "[.\n]*Saved access token"); *)
+                 assert (exit_code =*= Exit_code.ok));
+
+             (* login should fail on second call *)
+             let exit_code = Login_subcommand.main caps [| "semgrep-login" |] in
+             (* assert (res.logs =~ ".*You're already logged in"); *)
+             assert (exit_code =*= Exit_code.fatal));
+
+         (* clear login (by logging out) *)
+         let exit_code =
+           Logout_subcommand.main
+             (caps :> < Cap.stdout >)
+             [| "semgrep-logout" |]
+         in
+         (* assert (res.logs =~ ".*Logged out!"); *)
+         assert (exit_code =*= Exit_code.ok);
+
+         (* logout twice should work *)
+         let exit_code =
+           Logout_subcommand.main
+             (caps :> < Cap.stdout >)
+             [| "semgrep-logout" |]
+         in
+         (* assert (res.logs =~ ".*You are not logged in"); *)
```

Comment:
what about to those assert?

Reply:
This is now handled by capturing the output and comparing against the reference snapshot (in which everything is masked except for these strings we want to check).

I'll remove the asserts.

Reply:
https://github.com/semgrep/semgrep/pull/9762